### DISCUSSION TITLE
Create webapp that can load external plugins

### DISCUSSION THREAD

**Discussion by tardyp** [webpack/webpack#14448] [Q&A]
Hello,

maintainer of https://github.com/buildbot/buildbot
We are looking at redesigning our web front end with react from angular.js (1.x).

Buildbot is designed to be very configurable, and we want users to be able to create their own custom pages, which are then packaged in a .js and .css file that is just loaded inside the webapp (via script tags)

With angular.js, we could just provide library and UI code via angular.js's dependency injection, we just had to provide the plugin the angular object as a global variable and everything will just load and connect together.

We would like to reproduce this with webpack.

plugins can just import the buildbot library from their plugin. 
autocomplete should work and everything js developers are expecting. (This was a very loud complain from our users that autocomplete do not work with angular.js dependency injection)

```js
import React from 'react';
import BuildWidget from 'buildbot-base';
import registerMenuItem from 'buildbot-base';
```

buildbot-base code should not be included in the plugins bundle, but linked at runtime from the main app's bundle.
react and other base third party libraries should also be linked at runtime from the main app's bundle.

We can load plugins via script tags or via es6 dynamic import if that would simplify the design.

What do you think would be the best approach to this problem?

**Comment by alexander-akait** [comment]
Better to ask this on stackoverflow, because it is architecture design, and not related to webpack itself, we faced with this using any another bundle
